test/deals/post-deals.api-test.ts

```diff
@@ -0,0 +1,78 @@
+import { Api } from '@ukef-test/support/api';
+import { ENVIRONMENT_VARIABLES } from '@ukef-test/support/environment-variables';
+import { RandomValueGenerator } from '@ukef-test/support/generator/random-value-generator';
+import nock from 'nock';
+
+describe('POST /deals', () => {
+  const valueGenerator = new RandomValueGenerator();
+
+  const portfolioIdentifier = 'E1';
+  const idToken = valueGenerator.string();
+
+  const postDealUrl = `/api/v1/deals`;
+  const acbsDealResponse = {};
+  const expectedDealIdentifierResponse = { dealIdentifier: '00001' };
+
+  let api: Api;
+  beforeAll(async () => {
+    api = await Api.create();
+  });
+  afterAll(async () => {
+    await api.destroy();
+  });
+
+  it('returns a 201 response with the identifier of the new deal if ACBS responds with 201', async () => {
+    givenAuthenticationWithTheIdpSucceeds();
+    requestToCreateDealWithAcbs().reply(201, acbsDealResponse);
+
+    const { status, body } = await api.post(postDealUrl);
+
+    expect(status).toBe(201);
+    expect(body).toStrictEqual(JSON.parse(JSON.stringify(expectedDealIdentifierResponse)));
+  });
+
+  it('returns a 400 response if ACBS response with 400', async () => {
+    givenAuthenticationWithTheIdpSucceeds();
+    requestToCreateDealWithAcbs().reply(400, 'An error message from ACBS.');
+
+    const { status, body } = await api.post(postDealUrl);
+
+    expect(status).toBe(500);
+    expect(body).toStrictEqual({
+      statusCode: 500,
+      message: 'Internal server error',
+    });
+  });
```

Comment:
Can we have test where we send same payload twice, it is quite common situation. If it is not difficult.

Response should be 400, ACBS will return "Deal exists", we could share this error with caller, but should we?